File: android/app/src/main/java/com/dailymemory/domain/usecase/reminder/UpdateReminderUseCase.kt
package com.dailymemory.domain.usecase.reminder

import com.dailymemory.domain.model.Reminder
import com.dailymemory.domain.repository.ReminderRepository
import javax.inject.Inject

/**
 * Use case for updating an existing reminder.
 */
class UpdateReminderUseCase @Inject constructor(
    private val reminderRepository: ReminderRepository
) {
    suspend operator fun invoke(reminder: Reminder): Result<Reminder> {
        return try {
            reminderRepository.update(reminder)
            Result.success(reminder)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: ios/DailyMemory/Domain/UseCase/Reminder/UpdateReminderUseCase.swift
import Foundation

/// Use case for updating an existing reminder.
final class UpdateReminderUseCase {
    private let reminderRepository: ReminderRepository

    init(reminderRepository: ReminderRepository = ReminderRepositoryImpl()) {
        self.reminderRepository = reminderRepository
    }

    func execute(_ reminder: Reminder) async throws -> Reminder {
        try await reminderRepository.update(reminder)
        await SyncManager.shared.enqueueReminderChange(id: reminder.id, type: .update)
        return reminder
    }
}
